src/main/java/org/zalando/nakadi/service/subscription/StreamingContext.java

```diff
@@ -92,9 +92,20 @@ public long getKafkaPollTimeout() {
 
     @Override
     public void stream() throws InterruptedException {
+        // bugfix ARUHA-485
+        Runtime.getRuntime().addShutdownHook(new Thread(() -> onShutdown()));
```

Comment:
Actually, the target of the test is the function `onShutdown`. I would like to test it to be sure that if something changes in the future, it has a test to verify it.